### DISCUSSION TITLE
[GUIDE] How the priority system works

### DISCUSSION THREAD

**Discussion by Lailloken** [Lailloken/Exile-UI#17] [Announcements]
### General idea
The idea behind the prio-list is to set up a set of four archnemesis mods that you want to run "fully juiced," i.e. those big wombo-combos that you see on Reddit and such. The tool would then suggest a fast route to completing this list but would also suggest non-prioritized recipes and mods in case none of the prioritized recipes are ready.

### Mechanics
The tool generally tries to suggest the shortest path to completing this prio-set, i.e. it will not suggest completing a sub-recipe if you have already reached the required number of that mod. As an example:
- A set of Innocence-, Kitava-, and Shakari-touched requires 4 Gargantuans (2x emp. minions, 1x rejuvenating, 1x soul eater)
- you already have one emp. minions, so you'll only need an additional three Gargantuans
- if you drop one, the tool will put emp. minions (if you have the rest of the components ready) at the top of the ready-list and stop suggesting it once you complete it; from now on, all gargantuans will go to the other prio-recipes

The order in which recipes are added to the list also has an effect on how the tool makes suggestions. The first item on the list has its chain checked for available sub-components first, then the second, etc. This is also why recipes that have at least one assembled version present in the inventory are put at the end of the list automatically. 

### Bottlenecking
Being set up like this has its drawbacks, namely bottleneck-situations where there is a surplus in one branch and a dry-streak in another, which can lead to running out of inventory space. As a first step to counter this, I have implemented a panel that shows your surplus in mods connected to the priority list which can be used as a last resort to free up space.

**Comment by Hillgrove** [comment]
sounds amazing :)

**Comment by ItzBaka** [comment]
Thanks for your hard work!

**Comment by redrios** [comment]
you're my guy <3
